gprofiler/main.py

```diff
@@ -204,8 +204,18 @@ def start(self):
         self._stop_event.clear()
         self._system_metrics_monitor.start()
 
-        for prof in self.all_profilers:
-            prof.start()
+        for prof in list(self.all_profilers):
```

Comment:
Why did you add the `list`?

Reply:
`all_profilers` is a generator and we're now modifying the list this generator is based on (during iteration). So I first consume the generator to avoid modification during iteration